pkg/controller/manager/manager_controller.go

```diff
@@ -275,41 +275,35 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 
 	// If the manager TLS secret exists, check whether it is managed by the
 	// operator.
-	certOperatorManaged := true
-	if tlsSecret != nil {
-		certOperatorManaged, err = utils.IsCertOperatorIssued(tlsSecret.Data[render.ManagerInternalSecretCertName])
+	var operatorManagedCertSecret bool
+	if installation.CertificateManagement == nil {
+		// We use EnsureCertificateSecret to ensure a secret exists, creating one if one is not passed in.
+		// It also ensures the secret passed has the proper DNS names if the secret is operator managed.
+
+		svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
+		svcDNSNames = append(svcDNSNames, "localhost")
+		certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
+		tlsSecret, operatorManagedCertSecret, err = utils.EnsureCertificateSecret(
+			render.ManagerTLSSecretName, tlsSecret, render.ManagerSecretKeyName, render.ManagerSecretCertName, certDur, svcDNSNames...,
+		)
+
+		if err != nil {
+			r.status.SetDegraded(fmt.Sprintf("Error ensuring manager TLS certificate %q exists and has valid DNS names", render.ManagerTLSSecretName), err.Error())
+			return reconcile.Result{}, err
+		}
+
+	} else if tlsSecret != nil {
+		operatorManagedCertSecret, err = utils.IsCertOperatorIssued(tlsSecret.Data[render.ManagerInternalSecretCertName])
```

Comment:
The code here does not need to set the higher level scoped `operatorManagedCertSecret`, we haven't generated the secret here so no need to create it or put ownership on it.

Reply:
After chatting on a call, I think this comment should be ignored